core/src/test/scala/caliban/validation/InputObjectSpec.scala

```diff
@@ -139,6 +139,34 @@ object InputObjectSpec extends DefaultRunnableSpec {
                    )
                  )
         } yield assert(res.errors)(isEmpty)
+      },
+      testM("allow Int passed to Float field") {
+        val query =
+          """query QueryName($input: TestInputObjectInput!) {
+            |  query(input: $input)
+            |}""".stripMargin
+
+        case class TestInputObject(float: Float)
+        case class TestInput(input: TestInputObject)
+        case class TestOutput(value: String)
+        case class Query(query: TestInput => String)
+        val gql = graphQL(RootResolver(Query(_.input.float.toString)))
+
+        for {
+          int <- gql.interpreter
+          res <- int.executeRequest(
+                   GraphQLRequest(
+                     query = Some(query),
+                     variables = Some(
+                       Map(
+                         "input" -> InputValue.ObjectValue(
+                           Map("float" -> Value.IntValue(42))
```

Comment:
Hm, this strikes me as a bit odd since JSON only supports floats 🤔 I would probably have expected the coercion to be the other way around...

But perhaps we have some smartness in the parser to parse numbers without decimals as ints so we need to deal with both cases?

Reply:
The parser would parse into a `FloatValue` only if there is a dot or a `e` for exponent.

Reply:
In scala 2. The scala 3 one checks for int before float so the result is the same.

Reply:
According to the spec, float should not be coerced into int.